packages/database/src/core/util/libs/parser.ts

```diff
@@ -119,13 +138,21 @@ export const parseURL = function(
       dataURL = dataURL.substring(colonInd + 2);
     }
 
-    // Parse host and path.
+    // Parse host, path, and query string.
     let slashInd = dataURL.indexOf('/');
     if (slashInd === -1) {
       slashInd = dataURL.length;
     }
-    host = dataURL.substring(0, slashInd);
-    pathString = decodePath(dataURL.substring(slashInd));
+    let questionInd = dataURL.indexOf('?');
+    if (questionInd === -1) {
+      questionInd = dataURL.length;
+    }
+    host = dataURL.substring(0, Math.min(slashInd, questionInd));
+    if (slashInd < questionInd) {
+      // For pathString, questionInd will always come after slashInd
+      pathString = decodePath(dataURL.substring(slashInd, questionInd));
+    }
+    let queryParams = decodeQuery(dataURL.substring(Math.min(dataURL.length, questionInd + 1)));
```

Comment:
Can you let decodeQuery deal with the leading `?` the same way that decodePath does?

Reply:
done